### PR TITLE
Do not restart if update not needed

### LINKED ISSUE #740: "./bitwarden.sh update" restarts services even if there is no update
Is there a reason the entire Bitwarden stack is torn down when executing the `./bitwarden.sh update` command even if there are no updates available?  It takes about 5-6 minutes for Bitwarden to fully reinitialize and `dotnet` loads several cores at 100% during this time.

### PR DESCRIPTION
**Pull request by Mart124**
Hi,

This PR prevents Bitwarden to be restarted during the update process, if there's no update to perform.
This then closes #740.

Thx 👍

### CODE REVIEW

**Comment by Caligatio** [comment]
Did you mean to add the trailing colon in the second `grep`?  That colon causes the return code to be 1.

**Comment by Mart124** [comment]
@Caligatio no, because this trailing colon is also in the `docker` output :
`--format='{{.Config.Image}}:'`
This is a trick to properly catch to end of the version number, because `grep -F` can't catch it with `$`, so we catch it with an additional colon.

**Comment by Caligatio** [comment]
Ahh, I missed that colon in the format!  Please ignore my previous comment :)

File: scripts/run.sh [hunk 1]
```diff
@@ -185,6 +185,14 @@ then
     updateDatabase
 elif [ "$1" == "update" ]
 then
+    CORE_ID=$(docker-compose ps -q admin)
+    WEB_ID=$(docker-compose ps -q web)
+    if docker inspect --format='{{.Config.Image}}:' $CORE_ID | grep -F ":$COREVERSION:" | grep -q ":[0-9.]*:$" &&
+       docker inspect --format='{{.Config.Image}}:' $WEB_ID | grep -F ":$WEBVERSION:" | grep -q ":[0-9.]*:$"
+    then
```
[the remark this answers is not in the document]
Comment (Mart124) [reply scripts/run.sh:192]: They are excluded by the last `grep -q ":[0-9.]*:$` which will then return 1, making the whle condition to fail 👍

[the remark this answers is not in the document]
Comment (Caligatio) [reply scripts/run.sh:192]: The second `grep` only hits on versions that are solely comprised of periods and numbers. Versions that are words will always be restated.

**Comment by kspearrin** [comment]
Any way we can do something similar for powershell?

**Comment by Mart124** [comment]
Perhaps something "like" the following, but I don't have powershell to help 😕

```
$coreId = docker-compose ps -q admin
$webId = docker-compose ps -q web
if ((docker inspect --format='{{.Config.Image}}:' $coreId | Select-String :$coreVersion: | Select-String -Pattern ":[0-9.]*:$") && (docker inspect --format='{{.Config.Image}}:' $webId | Select-String :$webVersion: | Select-String -Pattern ":[0-9.]*:$")) {
     Write-Line "Update not needed"
     exit
}
```

**Pull request merged**